#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_no-check-nested-functions.dpatch
##                             by Rafael Laboissiere <rafael@debian.org>
##
## DP: Avoid global function definitions taking over local ones
## DP: (this patch is for octave2.9 only)

@DPATCH@

--- octave2.9-2.9.3.orig/src/variables.cc
+++ octave2.9-2.9.3/src/variables.cc
@@ -983,7 +983,12 @@
 {
   bool retval = false;

-  if (fcn)
+  // No need to check nested functions.  They can only be executed
+  // from within the parent function that contains them.  Parent
+  // and nested functions will be updated simultaneously when we
+  // check the parent.
+
+  if (fcn && ! fcn->is_nested_function ())
     {
       std::string ff = fcn->fcn_file_name ();

